lib/httpDigest.js

```diff
@@ -1,55 +1,100 @@
+ *
+ * @param {object} options - The options to use.
+ * @param {string|object} options.data - The data to be verified.
+ * @param {string} options.headerValue - The digest header value to verify
+ *   the data against.
+ *
+ * @returns {object} - An object with `verified` set to `true` or `false`.
+ */
+export async function verifyHeaderValue({data, headerValue} = {}) {
+  const {key, algorithm, encodedDigest} = _parseHeaderValue(headerValue);
+  const {encodedDigest: expectedDigest} = await _createHeaderValueComponents(
+    {data, algorithm, useMultihash: key === 'mh'});
+  return {verified: encodedDigest === expectedDigest};
+}
+
+async function _createHeaderValueComponents(
+  {data, algorithm = 'sha256', useMultihash = true} = {}) {
   if(typeof data !== 'string') {
     data = JSON.stringify(data);
   }
-  const encodedData = new TextEncoder().encode(data);
-  let digest;
+  if(algorithm !== 'sha256') {
+    throw new Error(`Algorithm "${algorithm}" is not supported.`);
+  }
+  const digest = await _getDigest({data, algorithm});
+  if(useMultihash) {
+    return {key: 'mh', encodedDigest: _createMultihash({digest})};
+  }
+  return {key: 'SHA-256', encodedDigest: base64Encode(digest)};
+}
 
-  switch(algorithm) {
-    case 'sha256': {
-      digest = new Uint8Array(
-        await crypto.subtle.digest({name: 'SHA-256'}, encodedData));
-      break;
-    }
-    case 'hs2019': {
-      // todo: support hs2019 / SHA-512
+function _parseHeaderValue(headerValue) {
+  const [key, ...rest] = headerValue.split('=');
+  const encodedDigest = rest.join('=');
+  let algorithm;
+  if(key === 'mh') {
+    // if `encodedDigest` starts with `uEi`, then it is a base64url-encoded
+    // sha-256 multihash
+    if(encodedDigest.startsWith('uEi')) {
+      algorithm = 'sha256';
+    } else {
+      throw new Error(
+        `Only base64url-encoded, sha-256 multihash is supported.`);
     }
-    default: {
-      throw new Error(`${algorithm} is not unsupported.`);
+  } else {
+    algorithm = key.replace('-', '').toLowerCase();
+    if(algorithm !== 'sha256') {
+      throw new Error(`Algorithm "${algorithm}" is not supported.`);
     }
   }
+  return {key, algorithm, encodedDigest};
+}
 
-  if(useMultihash) {
-    // format as multihash digest
-    // sha2-256: 0x12, length: 32 (0x20), digest value
-    const mh = new Uint8Array(34);
-    mh[0] = 0x12;
-    mh[1] = 0x20;
-    mh.set(digest, 2);
-    // encode multihash using multibase, base64url: `u`
-    return `mh=u${base64url.encode(mh)}`;
+async function _getDigest({data, algorithm}) {
+  const encodedData = new TextEncoder().encode(data);
+  if(algorithm === 'sha256') {
+    return new Uint8Array(
+      await crypto.subtle.digest({name: 'SHA-256'}, encodedData));
   }
+  throw new Error(`Algorithm "${algorithm}" is not unsupported.`);
+}
 
-  return `SHA-256=${base64.encode(digest)}`;
-};
-
-export default {
-  create
-};
+function _createMultihash({digest}) {
```

Comment:
would there be any benefit to using the npm packages for this and other multihash operations?

Reply:
@dlongley @dmitrizagidulin ^

Reply:
@mattcollier - it's a tough call, I always struggle with that. On the one hand, we should support all sorts of multihash algs, that's what multihash is _for_, right? But on the other, I'm always reluctant to pay the extra library size, especially for isomorphic libs.

@dlongley ?

Reply:
Don't want to pay for cost of the lib -- also don't know if the libs are isomorphic.